### PR TITLE
Add Intel oneAPI Toolkits

### PR DESCRIPTION
**Pull request by core-man**
Related to #90.

### CODE REVIEW

File: source/computer/fedora-setup.rst [hunk 1]
```diff
@@ -142,6 +142,19 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+
+`Intel oneAPI Toolkits <https://software.intel.com/content/www/us/en/develop/tools/oneapi.html>`__
+是英特尔最新的软件开发工具包。它也提供了 C/C++ 编译器和 Fortran 编译器。此外还有 MKL 数学库、MPI 并行库等。
+该工具包是免费的，不需要许可证。
+
+日常科研中安装 Base Toolkit 和 HPC Toolkit 即可，
+从官网\ `下载安装程序 <https://software.intel.com/content/www/us/en/develop/tools/oneapi/all-toolkits.html>`__\ 。
+下载时，选择 Local Installer，使用 ``wget`` 命令下载安装脚本。
+然后使用 ``sudo`` 运行安装脚本。工具包默认安装在 :file:`/opt/intel/oneapi` 目录下。
+安装完成后，按照指示设置好环境变量就可以使用该软件包了。
```
Comment (core-man) [inline source/computer/fedora-setup.rst:156]:
```suggestion
安装完成后，按照指示设置好环境变量就可以使用该工具包了。
```

File: source/computer/fedora-setup.rst [hunk 2]
```diff
@@ -142,6 +142,19 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+
+`Intel oneAPI Toolkits <https://software.intel.com/content/www/us/en/develop/tools/oneapi.html>`__
+是英特尔最新的软件开发工具包。它也提供了 C/C++ 编译器和 Fortran 编译器。此外还有 MKL 数学库、MPI 并行库等。
+该工具包是免费的，不需要许可证。
+
+日常科研中安装 Base Toolkit 和 HPC Toolkit 即可，
+从官网\ `下载安装程序 <https://software.intel.com/content/www/us/en/develop/tools/oneapi/all-toolkits.html>`__\ 。
+下载时，选择 Local Installer，使用 ``wget`` 命令下载安装脚本。
+然后使用 ``sudo`` 运行安装脚本。工具包默认安装在 :file:`/opt/intel/oneapi` 目录下。
+安装完成后，按照指示设置好环境变量就可以使用该工具包了。
```
Comment (seisman) [inline source/computer/fedora-setup.rst:156]: why not use the `yum` or `dnf`?

https://software.intel.com/content/www/us/en/develop/articles/installing-intel-oneapi-toolkits-via-yum.html

Comment (core-man) [reply source/computer/fedora-setup.rst:156]: I did not try it. Intel recommend local installer. Will try it.

Comment (core-man) [reply source/computer/fedora-setup.rst]: We can use apt and dnf for Ubuntu and Fedora. For macOS, we can only use local installer.

See ed48372ad8c9fe712a96796717ee8c1d2ffbbbd8.

File: source/computer/fedora-setup.rst [hunk 3]
```diff
@@ -142,6 +142,38 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+生成 :file:`.repo` 文件，命名为 :file:`oneAPI.repo`，并放到 :file:`/etc/yum.repos.d` 目录下::
+
+    $ cat /etc/yum.repos.d/oneAPI.repo
+    [oneAPI]
+    name=Intel(R) oneAPI repository
+    baseurl=https://yum.repos.intel.com/oneapi
+    enabled=1
+    gpgcheck=1
+    repo_gpgcheck=1
+    gpgkey=https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
```
Comment (seisman) [inline source/computer/fedora-setup.rst:165]: I'm not sure if users really know how to create a file in `/etc/yum.repos.d` directory.

Comment (core-man) [reply source/computer/fedora-setup.rst:165]: We can create such a file and let babies copy to `/etc/yum.repos.d/`?

File: source/computer/ubuntu-setup.rst [hunk 4]
```diff
@@ -141,6 +141,42 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+
+添加 Intel 仓库公钥::
+
+    # 下载 Intel 仓库公钥
+    cd ~/Downloads
+    wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+    # 添加到 apt 源密钥环中，以便使用此密钥签名的档案受到信任
+    sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+    # 删除公钥
+    rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
```
Comment (seisman) [inline source/computer/ubuntu-setup.rst:163]:
```suggestion
    # 下载 Intel 仓库公钥
    $ cd ~/Downloads
    $ wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
    # 添加到 apt 源密钥环中，以便使用此密钥签名的档案受到信任
    $ sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
    # 删除公钥
    $ rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
```

File: source/computer/fedora-setup.rst (hunk 3 above)
Comment (seisman) [reply source/computer/fedora-setup.rst:165]: Yes, that's a better solution.

File: source/computer/fedora-setup.rst [hunk 5]
```diff
@@ -142,6 +142,38 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+安装工具包，默认安装目录是 :file:`/opt/intel/oneapi`::
+
+    $ sudo dnf install intel-basekit
+    $ sudo dnf install intel-hpckit
+
+安装完成后，按照\ `官方手册 <https://software.intel.com/content/www/us/en/develop/documentation/get-started-with-intel-oneapi-base-linux/>`__\
+配置好环境变量，就可以使用该工具包了::
+
+    $ echo "source /opt/intel/oneapi/setvars.sh >/dev/null 2>&1" >> ~/.bashrc
+
```
Comment (seisman) [inline source/computer/fedora-setup.rst:176]: It's a little confusing. Do I need to follow the "official tutorial" first and then change the `~/.bashrc`? Or I only need to update `~/.bashrc`?

File: source/computer/fedora-setup.rst (hunk 3 above)
Comment (core-man) [reply source/computer/fedora-setup.rst]: done in 887bf41a9a22797e48da3fd0497a4fc2866ad9b9.

File: source/computer/fedora-setup.rst (hunk 5 above)
Comment (core-man) [reply source/computer/fedora-setup.rst:169]: We only need to update `~/.bashrc` to use `icc` and `ifort`, but the official tutorial has more options.

Comment (core-man) [reply source/computer/fedora-setup.rst]: Clarify it in 570799a.

File: source/computer/fedora-setup.rst [hunk 6]
```diff
@@ -142,6 +142,32 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+下载 :file:`.repo` 文件 :download:`oneAPI.repo`\ ，并将其放在 :file:`/etc/yum.repos.d` 目录下::
+
+    $ sudo mv oneAPI.repo /etc/yum.repos.d/
+
+安装工具包，默认安装目录是 :file:`/opt/intel/oneapi`::
+
+    $ sudo dnf install intel-basekit
+    $ sudo dnf install intel-hpckit
+
```
Comment (seisman) [inline source/computer/fedora-setup.rst:164]:
```suggestion
在 Fedora 系统下，官方手册提供了多种\
`安装方式 <https://software.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top.html>`__\ ，
如在线安装、本地安装、使用 ``dnf`` 安装、使用 ``conda`` 安装等。这里，我们选择使用 ``dnf`` 安装。

下载 :file:`.repo` 文件 :download:`oneAPI.repo`\ ，并将其放在 :file:`/etc/yum.repos.d` 目录下::

    $ sudo mv oneAPI.repo /etc/yum.repos.d/

安装工具包，日常科研安装 Base Toolkit 和 HPC Toolkit 即可。默认安装目录是 :file:`/opt/intel/oneapi`::

    $ sudo dnf install intel-basekit
    $ sudo dnf install intel-hpckit

```

File: source/computer/fedora-setup.rst [hunk 7]
```diff
@@ -142,6 +142,32 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+在 Fedora 系统下，官方手册提供了多种\
+`安装方式 <https://software.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top.html>`__\ ，
+如在线安装、本地安装、使用 ``dnf`` 安装、使用 ``conda`` 安装等。这里，我们选择使用 ``dnf`` 安装。
+
+下载 :file:`.repo` 文件 :download:`oneAPI.repo`\ ，并将其放在 :file:`/etc/yum.repos.d` 目录下::
+
+    $ sudo mv oneAPI.repo /etc/yum.repos.d/
+
+安装工具包，日常科研安装 Base Toolkit 和 HPC Toolkit 即可。默认安装目录是 :file:`/opt/intel/oneapi`::
```
Comment (core-man) [inline source/computer/fedora-setup.rst:160]:
```suggestion
日常科研安装 Base Toolkit 和 HPC Toolkit 两个工具包即可。默认安装目录是 :file:`/opt/intel/oneapi`::
```

**Review by seisman — APPROVED** [review]
OK to merge after you apply the same fix to Ubuntu and macOS.

File: source/computer/macos-setup.rst [hunk 8]
```diff
@@ -192,6 +192,26 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+
+`Intel oneAPI Toolkits <https://software.intel.com/content/www/us/en/develop/tools/oneapi.html>`__
+是英特尔最新的软件开发工具包。它也提供了 C/C++ 编译器和 Fortran 编译器（``icc`` 和 ``ifort`` 命令）。
+此外还有 MKL 数学库、MPI 并行库等。该工具包是免费的，不需要许可证。
+
+日常科研安装 Base Toolkit 和 HPC Toolkit 即可。在 macOS 下，官方手册提供了多种\
+`安装方式 <https://software.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-macos/>`__\ ，
+如在线安装、本地安装、使用 ``conda`` 安装等。这里，我们选择本地安装。
+
+从官网\ `下载安装程序 <https://software.intel.com/content/www/us/en/develop/tools/oneapi/all-toolkits.html>`__\ 。
```
Comment (core-man) [inline source/computer/macos-setup.rst:206]:
```suggestion
日常科研安装 Base Toolkit 和 HPC Toolkit 两个工具包即可。从官网下载\ `安装程序 <https://software.intel.com/content/www/us/en/develop/tools/oneapi/all-toolkits.html>`__\ 。
```

File: source/computer/macos-setup.rst [hunk 9]
```diff
@@ -192,6 +192,26 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
     $ brew install gfortran
 
+Intel 软件开发工具包
+^^^^^^^^^^^^^^^^^^^^
+
+`Intel oneAPI Toolkits <https://software.intel.com/content/www/us/en/develop/tools/oneapi.html>`__
+是英特尔最新的软件开发工具包。它也提供了 C/C++ 编译器和 Fortran 编译器（``icc`` 和 ``ifort`` 命令）。
+此外还有 MKL 数学库、MPI 并行库等。该工具包是免费的，不需要许可证。
+
+日常科研安装 Base Toolkit 和 HPC Toolkit 即可。在 macOS 下，官方手册提供了多种\
```
Comment (core-man) [inline source/computer/macos-setup.rst:202]:
```suggestion
在 macOS 下，官方手册提供了多种\
```

File: source/computer/ubuntu-setup.rst [hunk 10]
```diff
@@ -141,6 +141,43 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
+    $ sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+    # 删除公钥
+    $ rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+
+添加 Intel 软件仓库::
+
+    $ sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
+    $ sudo apt update
+
+安装工具包，默认安装目录是 :file:`/opt/intel/oneapi`::
```
Comment (core-man) [inline source/computer/ubuntu-setup.rst:170]:
```suggestion
日常科研安装 Base Toolkit 和 HPC Toolkit 两个工具包即可。默认安装目录是 :file:`/opt/intel/oneapi`::
```

File: source/computer/ubuntu-setup.rst [hunk 11]
```diff
@@ -141,6 +141,43 @@ Fortran 编译器，其提供了 ``gfortran`` 命令::
...
     $ sudo apt install gfortran
 
+Intel 软件开发工具包
+^^^^^^^^^^^^^^^^^^^^
+
+`Intel oneAPI Toolkits <https://software.intel.com/content/www/us/en/develop/tools/oneapi.html>`__
+是英特尔最新的软件开发工具包。它也提供了 C/C++ 编译器和 Fortran 编译器（``icc`` 和 ``ifort`` 命令）。
+此外还有 MKL 数学库、MPI 并行库等。该工具包是免费的，不需要许可证。
+
+日常科研安装 Base Toolkit 和 HPC Toolkit 即可。在 Ubuntu 系统下，官方手册提供了多种\
```
Comment (core-man) [inline source/computer/ubuntu-setup.rst:151]:
```suggestion
在 Ubuntu 系统下，官方手册提供了多种\
```

**Pull request merged**